legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/ProjectsVersionsResource.java

```diff
@@ -58,7 +59,17 @@ public Optional<ProjectVersionDTO> getProjectVersion(@PathParam("groupId") Strin
     {
         return handle(ResourceLoggingAndTracing.GET_PROJECT_VERSION_BY_GAV, ResourceLoggingAndTracing.GET_PROJECT_VERSION_BY_GAV + groupId + artifactId + versionId, () ->
         {
-            Optional<StoreProjectVersionData> projectVersion = projectVersionApi.find(groupId, artifactId, versionId);
+            String finalVersionId = versionId;
+            if (versionId.equalsIgnoreCase("latest"))
+            {
+                Optional<VersionId> latestVersion = projectVersionApi.getLatestVersion(groupId, artifactId);
```

Comment:
any logic to figure out latest version shuould be in the project service api. you need anew metho called getLatestVersionData

Reply:
getLatestVersion method is implemented in the project services impl itself, but we dont use the functionality currently anywhere to get latest version data , so I did not find it beneficial for it to be added to the service impl itself for just the api

Reply:
rest end points  should not have any business logic that's the responsibility of the services. So we need to add a new api in the service to get latest version data.
the criteria here is to have business logic isolated in the services not how many times we use the api